Source/WebKit/WebProcess/WebPage/Cocoa/WebRemoteObjectRegistry.h

```diff
@@ -26,31 +26,32 @@
 #pragma once
 
 #include "RemoteObjectRegistry.h"
+#include <wtf/RefCounted.h>
 #include <wtf/TZoneMalloc.h>
 #include <wtf/WeakRef.h>
 
-namespace WebKit {
-class WebRemoteObjectRegistry;
-}
-
-namespace WTF {
-template<typename T> struct IsDeprecatedWeakRefSmartPointerException;
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebKit::WebRemoteObjectRegistry> : std::true_type { };
-}
-
 namespace WebKit {
 
 class WebPage;
 
-class WebRemoteObjectRegistry final : public RemoteObjectRegistry {
+class WebRemoteObjectRegistry final : public RemoteObjectRegistry, public RefCounted<WebRemoteObjectRegistry> {
     WTF_MAKE_TZONE_ALLOCATED(WebRemoteObjectRegistry);
 public:
-    WebRemoteObjectRegistry(_WKRemoteObjectRegistry *, WebPage&);
+    static Ref<WebRemoteObjectRegistry> create(_WKRemoteObjectRegistry *remoteObjectRegistry, WebPage& page)
+    {
+        return adoptRef(*new WebRemoteObjectRegistry(remoteObjectRegistry, page));
+    }
+
     ~WebRemoteObjectRegistry();
+
+    void ref() const final { RefCounted::ref(); }
+    void deref() const final { RefCounted::deref(); }
     
     void close();
     
 private:
+    WebRemoteObjectRegistry(_WKRemoteObjectRegistry *, WebPage&);
+
     MessageSender messageSender() final;
     uint64_t messageDestinationID() final;
 
```

Comment:
Well, same old bug. The WeakRef below needs to become a WeakPtr with some null check since you're extending lifetime.

Reply:
Fixed!